### PR TITLE
Use mobile web content for iPad and desktop for mac

### PR DESCRIPTION
**Pull request by bgoncal**
## Link to pull request in Documentation repository

Documentation: home-assistant/companion.home-assistant#

### CODE REVIEW

**Comment by bgoncal** [comment]
On hold while discussing #2908

File: Sources/App/WebView/WebViewController.swift [hunk 1]
```diff
@@ -136,6 +136,7 @@ final class WebViewController: UIViewController, WKNavigationDelegate, WKUIDeleg
 
         config.userContentController = userContentController
         config.applicationNameForUserAgent = HomeAssistantAPI.applicationNameForUserAgent
+        config.defaultWebpagePreferences.preferredContentMode = Current.isCatalyst ? .desktop : .mobile
```
Comment (steverep) [inline Sources/App/WebView/WebViewController.swift:139]: If desktop is still used for Mac, I think the resulting user agent string would still unfortunately result in always serving the legacy frontend build, correct?

Comment (bgoncal) [reply Sources/App/WebView/WebViewController.swift:139]: If I set "mobile" to be used for Mac the result user agent is even worse:

`Mozilla/5.0 (iPad; CPU OS 14_5 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Home Assistant/2024.8 (io.robbie.HomeAssistant.dev; build:2024; macOS 14.5.0) Mobile/HomeAssistant, like Safari`

Says "iPad" and "OS 14_5" which is not the iPad version (it's macOS version)

Comment (steverep) [reply Sources/App/WebView/WebViewController.swift:139]: Ugh, yeah that's terrible.  Like I said, Apple's fault.

Comment (bgoncal) [reply Sources/App/WebView/WebViewController.swift:139]: Indeed and looking at historic data, they are not working to improve this 😅

**Comment by bgoncal** [comment]
Now we need a solution for macOS

**Pull request merged**